src/components/BlogForm/NewPost.tsx

```diff
@@ -0,0 +1,117 @@
+"use client"; // needed for useform
+import { Button, Stack, TextField, Typography } from "@mui/material";
+import { useForm, Controller } from "react-hook-form";
+import { postBlog } from "../../api/blog";
+import { useSession } from "next-auth/react";
+import { DataProps } from "./types";
+
+const NewPost = () => {
+  const { data: session } = useSession();
+  const user = session?.user.name as string;
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      author: "",
+      title: "",
+      body: "",
+      date: "",
+      tags: "",
+    },
+  });
+
+  return (
+    <form
+      onSubmit={handleSubmit((formdata) => {
+        AddAndSubmit(formdata, user);
```

Comment:
this feels clunky, can you rename to onSubmit and move inside the component function to call the user variable directly. then just pass in

`handleSubmit(onSubmit)`